narwhals/expression.py

```diff
@@ -2081,6 +2081,92 @@ def to_datetime(self, format: str) -> Expr:  # noqa: A002
             lambda plx: self._expr._call(plx).str.to_datetime(format=format)
         )
 
+    def to_uppercase(self) -> Expr:
+        r"""
+        Transform string to uppercase variant.
+
+        Notes:
+            The PyArrow backend will convert 'ß' to 'ẞ' instead of 'SS'.
+            For more info see [the related issue](https://github.com/apache/arrow/issues/34599).
+            There may be other unicode-edge-case-related variations across implementations.
```

Comment:
Not sure how much we want to expand on this topic, but I would consider adding some additional information, such as mentioning that pyarrow is based on [utf8proc](https://juliastrings.github.io/utf8proc/) for sake of completeness.

WDYT?

Reply:
sure, happy to take this as a follow-up! 👍 for now I'll merge as-is